### ISSUE TITLE
Error in lowering tf.ResizeBilinear and tf.ResizeNearestNeighbor using iree-import-tf

### ISSUE THREAD

**Issue by ArunaKote** [google/iree#7045]
Model  for tf.ResizeNearestNeighbor  looks like:
model = tf.keras.Sequential([
     tf.keras.layers.Conv2D(4, (1, 1),input_shape = (10, 10, 3),batch_size=1, name='fpn_c5p5'),
     tf.keras.layers.UpSampling2D(size=(2, 2),interpolation='nearest', name="fpn_p5upsampled")
])

Model  for  tf.ResizeBilinear  looks like:
model = tf.keras.Sequential([
     tf.keras.layers.Conv2D(4, (1, 1),input_shape = (10, 10, 3),batch_size=1, name='fpn_c5p5'),
     tf.keras.layers.UpSampling2D(size=(2, 2),interpolation='nearest', name="fpn_p5upsampled")
])

when I tried to lower using iree-import-tf getting error: for  tf.ResizeBilinear:
<unknown>:0: error: loc(callsite(callsite("sequential/fpn_p5upsampled/resize/ResizeBilinear@__inference__wrapped_model_53" at "StatefulPartitionedCall@__inference_signature_wrapper_195") at "StatefulPartitionedCall")): 'tf.ResizeBilinear' op : unlegalized TensorFlow op still exists
<unknown>:0: note: loc("StatefulPartitionedCall"): called from
<unknown>:0: note: loc(callsite(callsite("sequential/fpn_p5upsampled/resize/ResizeBilinear@__inference__wrapped_model_53" at "StatefulPartitionedCall@__inference_signature_wrapper_195") at "StatefulPartitionedCall")): see current operation: %6 = "tf.ResizeBilinear"(%5, %0) {align_corners = false, device = "", half_pixel_centers = true} : (tensor<1x10x10x4xf32>, tensor<2xi32>) -> tensor<1x20x20x4xf32>
<unknown>:0: error: The following Tensorflow operations still remain:
        tf.ResizeBilinear (count: 1)

for  tf.ResizeBilinear:
<unknown>:0: error: loc(callsite(callsite("sequential/fpn_p5upsampled/resize/tf.ResizeNearestNeighbor @__inference__wrapped_model_53" at "StatefulPartitionedCall@__inference_signature_wrapper_195") at "StatefulPartitionedCall")): 'tf.ResizeNearestNeighbor ' op : unlegalized TensorFlow op still exists
<unknown>:0: note: loc("StatefulPartitionedCall"): called from
<unknown>:0: note: loc(callsite(callsite("sequential/fpn_p5upsampled/resize/tf.ResizeNearestNeighbor @__inference__wrapped_model_53" at "StatefulPartitionedCall@__inference_signature_wrapper_195") at "StatefulPartitionedCall")): see current operation: %6 = "tf.tf.ResizeNearestNeighbor "(%5, %0) {align_corners = false, device = "", half_pixel_centers = true} : (tensor<1x10x10x4xf32>, tensor<2xi32>) -> tensor<1x20x20x4xf32>
<unknown>:0: error: The following Tensorflow operations still remain:
        tf.ResizeNearestNeighbor  (count: 1)
[resize.zip](https://github.com/google/iree/files/7168722/resize.zip)

The attachment contains python files and log files for both.

**Comment by GMNGeoffrey** [comment]
Hi Aruna,

Responding here to this and the other issues you filed. Thank you for your detailed bug reports. Unfortunately, supporting the full scope of the TF API with all its many weirdnesses is not something we are staffed or prepared to do at the moment. Almost all of your bug reports are at the level of converting TF -> MHLO. I understand that from a user perspective, where the thing fails isn't super important, but for us it means that this solidly falls into the category of "TF rationalizing its frontend". The actual code for doing that lives in the [TF repository](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/mlir/xla) and iree-import-tf mostly just hooks into it in a certain way to provide a standalone tool that we can actually use. We deliberately limit the amount that we wade into that space (although we do sometimes for unblocking things particularly important to us). So while I would love to help you, I think these issues should likely be filed in the TensorFlow repository, although I suspect that they might also find limited usefulness in individual issues that basically are "lowering of this op isn't implemented". I suspect they are aware that not all lowerings are implemented and have some kind of work tracking/prioritization/burndown list. Letting them know that there are certain ops that are important to you and aren't getting lowered might be useful, however. If you find that the TF->MHLO lowering for the ops being reported *do* exist, and the IREE TF import pipeline isn't picking them up for some reason, that would likely be a bug in IREE that we can investigate.

In the meantime, we can do a better job documenting the TF support status, but to quote from our README as a general point.

> [IREE] is still quite far from ready for everyday use and is made available without any support at the moment.

Best,
Geoffrey

**Comment by ArunaKote** [comment]
Thank You

**Comment by dinkdeep** [comment]
Hi Geoffrey ,
Irrespective of iree if something cannot be lowered from TF->HLO then it cannot be further lowered  to linalg so essentially it cannot go through the MLIR path end to end, But on the other hand the used cases seem to work with the XLA path, Ideally isn't the goal that any thing which works on XLA path should also work on the MLIR, Given said may be if you can pitch in key persons and point out there names we can approach them and work with them if needed.

**Comment by GMNGeoffrey** [comment]
Yeah that sounds correct in general, with the nit that there is no "the MLIR path end to end", but rather many paths from various ends to various other ends that leverage the MLIR infrastructure. But assuming you are referring to the TF->XLA bridge, then I agree that this seems like a deficiency.

@smit-hinsu could you help these folks find the right place to catalogue these issues?

**Comment by jpienaar** [comment]
@ArunaKote are you running this with xla-legalize-tf-with-tf2xla option enabled? (seems this op is only supported on fallback lowering). @paynecl

**Comment by smit-hinsu** [comment]
CC @mikedelorimier

**Comment by ArunaKote** [comment]
@jpienaar , I have tried xla-legalize-tf-with-tf2xla option enabled. I am not able to lower .When I saw source code i.e https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/image_resize_ops.cc I found 
ResizeNearestNeighbor with half_pixel_centers=True is not yet implemented and ResizeNearestNeighbor with align_corners=False is not yet implemented.
I have tried with xla-leagalize-tf ,If align_corners= false,half_pixel_centers=false then I able to lower it. But if it has align_corners = false,  half_pixel_centers = true i am not able to lower it.

**Comment by smit-hinsu** [comment]
We can revive  the following PR for ResizeNearestNeighbor op when half_pixel_centers=true.

https://github.com/tensorflow/tensorflow/pull/46836

**Comment by GMNGeoffrey** [comment]
I can't transfer these issues to the TF repository (only possible within a GitHub organization). I'm going to close this and related issues as out of scope for IREE. TF folks, can you please revive these as TF issues where appropriate? Basically all of these: https://github.com/google/iree/issues?q=is%3Aissue+author%3AArunaKote+sort%3Aupdated-desc+

**Comment by GMNGeoffrey** [comment]
Closed #7050, #7051, #7052, #7046, #7047, #7048, #7049, #6893, #6920, #6956. Please reopen if you think this was an error
